Setting Arducam Model on Camera tab fails

**Describe the bug**
Trying to set the Arducam Model for a camera on the Camera tab fails with a message "Failed to process camera settings request" in the UI.

**Screenshots / Videos**
The logs contain the following
```
[2026-07-13 03:04:18] [WebServer - Server] [DEBUG] Handled HTTP request of type POST from endpoint /api/settings/camera of req size 421 bytes & type application/json with return code 500 for host 192.168.50.29 in 2.165896 ms
[2026-07-13 03:06:19] [WebServer - RequestHandler] [ERROR] Failed to process camera settings request: No JsonAdapter for java.util.HashMap<org.photonvision.vision.camera.CameraQuirk, java.lang.Boolean>
Possible Causes:
1. Missing @Json or @Json.Import annotation.
2. The avaje-jsonb-generator dependency was not available during compilation or -proc full was not enabled in maven compiler
for java.util.HashMap<org.photonvision.vision.camera.CameraQuirk, java.lang.Boolean>
for class org.photonvision.server.RequestHandler$CameraSettingsRequest
[2026-07-13 03:06:19] [WebServer - RequestHandler] [ERROR] java.lang.IllegalArgumentException: No JsonAdapter for java.util.HashMap<org.photonvision.vision.camera.CameraQuirk, java.lang.Boolean>
Possible Causes:
1. Missing @Json or @Json.Import annotation.
2. The avaje-jsonb-generator dependency was not available during compilation or -proc full was not enabled in maven compiler
for java.util.HashMap<org.photonvision.vision.camera.CameraQuirk, java.lang.Boolean>
for class org.photonvision.server.RequestHandler$CameraSettingsRequest
        at io.avaje.jsonb.core.CoreAdapterBuilder$LookupChain.exceptionWithLookupStack(CoreAdapterBuilder.java:239)
        at io.avaje.jsonb.core.CoreAdapterBuilder.lookupAdapter(CoreAdapterBuilder.java:121)
        at io.avaje.jsonb.core.CoreAdapterBuilder.build(CoreAdapterBuilder.java:140)
        at io.avaje.jsonb.core.DJsonb.adapter(DJsonb.java:207)
        at org.photonvision.server.RequestHandler$CameraSettingsRequestJsonAdapter.<init>(RequestHandler$CameraSettingsRequestJsonAdapter.java:34)
        at io.avaje.jsonb.core.DJsonb$DBuilder.lambda$newAdapterFactory$2(DJsonb.java:415)
        at io.avaje.jsonb.core.CoreAdapterBuilder.lookupAdapter(CoreAdapterBuilder.java:98)
        at io.avaje.jsonb.core.CoreAdapterBuilder.build(CoreAdapterBuilder.java:140)
        at io.avaje.jsonb.core.DJsonb.adapter(DJsonb.java:207)
        at io.avaje.jsonb.core.DJsonb.lambda$typeWithCache$0(DJsonb.java:180)
        at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1724)
        at io.avaje.jsonb.core.DJsonb.typeWithCache(DJsonb.java:180)
        at io.avaje.jsonb.core.DJsonb.type(DJsonb.java:164)
        at org.photonvision.server.RequestHandler.onCameraSettingsRequest(RequestHandler.java:404)
        ...
```

**Platform:**
 - Hardware Platform (ex. Raspberry Pi 4, Windows x64): Orange Pi 5
   - How is it powered? (ex. Zinc-V, Pololu Buck Converter, Battery Bank): Wall wart
 - Network Configuration (Connection between the Radio and any devices in between, such as a Network Switch): Ethernet to router
 - PhotonVision Version: dev-v2027.0.0-alpha-2-18-g97fb6eab (from #2557)
 - Browser (with Version) (Chrome, Edge, Firefox, etc.): Edge
 - Camera(s) Used: Arducam OV9281

--- a/photon-client/tests/quirks.spec.ts
+++ b/photon-client/tests/quirks.spec.ts
@@ -1,0 +1,22 @@
+import { expect } from "@playwright/test";
+import { test } from "./fixtures.ts";
+
+test("Quirks should be able to be changed", async ({ page }) => {
+  await page.goto("http://localhost:5800/#/cameras");
+
+  await page.locator("div.d-flex", { has: page.locator("span", { hasText: "Arducam Model" }) }).locator("div.v-field").click();
+
+  await page.locator(".v-overlay .v-list .v-list-item", { hasText: "OV9281" }).click();
+
+  await page.locator("button", { has: page.locator("span", { hasText: "Save Changes" }) }).click();
+
+  await expect(page.locator(".v-overlay p").last()).toHaveText("Camera settings updated successfully");
+
+  await page.locator("div.d-flex", { has: page.locator("span", { hasText: "Arducam Model" }) }).locator("div.v-field").click();
+
+  await page.locator(".v-overlay .v-list .v-list-item", { hasText: "None" }).click();
+
+  await page.locator("button", { has: page.locator("span", { hasText: "Save Changes" }) }).click();
+
+  await expect(page.locator(".v-overlay p").last()).toHaveText("Camera settings updated successfully");
+});

--- a/photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java
+++ b/photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java
@@ -215,8 +215,8 @@ public class NetworkTablesManager {
     }
 
     private void broadcastConnectedStatusImpl() {
-        HashMap<String, Object> map = new HashMap<>();
-        var subMap = new HashMap<String, Object>();
+        Map<String, Object> map = new HashMap<>();
+        Map<String, Object> subMap = new HashMap<>();
 
         subMap.put("connected", ntInstance.isConnected());
         if (ntInstance.isConnected()) {

--- a/photon-core/src/main/java/org/photonvision/common/dataflow/websocket/UIDataPublisher.java
+++ b/photon-core/src/main/java/org/photonvision/common/dataflow/websocket/UIDataPublisher.java
@@ -19,6 +19,8 @@ package org.photonvision.common.dataflow.websocket;
 
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import org.photonvision.common.dataflow.CVPipelineResultConsumer;
 import org.photonvision.common.dataflow.DataChangeService;
 import org.photonvision.common.dataflow.events.OutgoingUIEvent;
@@ -46,11 +48,11 @@ public class UIDataPublisher implements CVPipelineResultConsumer {
         // only update the UI at 10hz
         if (lastUIResultUpdateTime + 1000.0 / 10.0 > now) return;
 
-        var dataMap = new HashMap<String, Object>();
+        Map<String, Object> dataMap = new HashMap<>();
         dataMap.put("sequenceID", result.sequenceID);
         dataMap.put("fps", result.fps);
         dataMap.put("latency", result.getLatencyMillis());
-        var uiTargets = new ArrayList<HashMap<String, Object>>(result.targets.size());
+        List<Map<String, Object>> uiTargets = new ArrayList<>(result.targets.size());
 
         // We don't actually need to send targets during calibration and it can take up a lot (up to
         // 1.2Mbps for 60 snapshots) of target results with no pitch/yaw/etc set
@@ -75,7 +77,7 @@ public class UIDataPublisher implements CVPipelineResultConsumer {
             dataMap.put("multitagResult", multitagData);
         }
 
-        var uiMap = new HashMap<String, HashMap<String, Object>>();
+        Map<String, Map<String, Object>> uiMap = new HashMap<>();
         uiMap.put(uniqueName, dataMap);
 
         if (result instanceof FocusPipelineResult focusResult) {

--- a/photon-core/src/main/java/org/photonvision/common/logging/Logger.java
+++ b/photon-core/src/main/java/org/photonvision/common/logging/Logger.java
@@ -307,10 +307,10 @@ public class Logger {
     private static class UILogAppender implements LogAppender {
         @Override
         public void log(String message, LogLevel level) {
-            var messageMap = new HashMap<String, Object>();
+            Map<String, Object> messageMap = new HashMap<>();
             messageMap.put("logMessage", message);
             messageMap.put("logLevel", level.code);
-            var superMap = new HashMap<String, Object>();
+            Map<String, Object> superMap = new HashMap<>();
             superMap.put("logMessage", messageMap);
             DataChangeService.getInstance().publishEvent(OutgoingUIEvent.wrappedOf("log", superMap));
         }

--- a/photon-core/src/main/java/org/photonvision/vision/pipeline/Calibrate3dPipeline.java
+++ b/photon-core/src/main/java/org/photonvision/vision/pipeline/Calibrate3dPipeline.java
@@ -20,6 +20,7 @@ package org.photonvision.vision.pipeline;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import org.opencv.core.Mat;
 import org.opencv.core.Point;
 import org.photonvision.common.dataflow.DataChangeService;
@@ -190,7 +191,7 @@ public class Calibrate3dPipeline
     }
 
     public void broadcastState() {
-        var state =
+        Map<String, Object> state =
                 SerializationUtils.objectToHashMap(
                         new UICalibrationData(
                                 foundCornersList.size(),

--- a/photon-core/src/main/java/org/photonvision/vision/processes/VisionModule.java
+++ b/photon-core/src/main/java/org/photonvision/vision/processes/VisionModule.java
@@ -751,7 +751,7 @@ public class VisionModule {
      *
      * @param quirksToChange map of true/false for quirks we should change
      */
-    public void changeCameraQuirks(HashMap<CameraQuirk, Boolean> quirksToChange) {
+    public void changeCameraQuirks(Map<CameraQuirk, Boolean> quirksToChange) {
         visionSource.getCameraConfiguration().cameraQuirks.updateQuirks(quirksToChange);
         visionSource.remakeSettables();
         saveAndBroadcastAll();

--- a/photon-core/src/main/java/org/photonvision/vision/processes/VisionModule.java
+++ b/photon-core/src/main/java/org/photonvision/vision/processes/VisionModule.java
@@ -544,8 +544,8 @@ public class VisionModule {
         logger.trace("Broadcasting PSC mutation - " + propertyName + ": " + value);
         saveModule();
 
-        HashMap<String, Object> map = new HashMap<>();
-        HashMap<String, Object> subMap = new HashMap<>();
+        Map<String, Object> map = new HashMap<>();
+        Map<String, Object> subMap = new HashMap<>();
         subMap.put(propertyName, value);
         map.put("mutatePipelineSettings", subMap);
 

--- a/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
+++ b/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
@@ -30,6 +30,7 @@ import java.nio.file.StandardCopyOption;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.LinkedList;
+import java.util.Map;
 import java.util.Optional;
 import javax.imageio.ImageIO;
 import org.apache.commons.io.FileUtils;
@@ -396,7 +397,7 @@ public class RequestHandler {
 
     @Json
     record CameraSettingsRequest(
-            double fov, HashMap<CameraQuirk, Boolean> quirksToChange, String cameraUniqueName) {}
+            double fov, Map<CameraQuirk, Boolean> quirksToChange, String cameraUniqueName) {}
 
     public static void onCameraSettingsRequest(Context ctx) {
         try {
@@ -404,7 +405,7 @@ public class RequestHandler {
                     Jsonb.instance().type(CameraSettingsRequest.class).fromJson(ctx.body());
             // Extract the settings from the request
             double fov = request.fov;
-            HashMap<CameraQuirk, Boolean> quirksToChange = request.quirksToChange;
+            Map<CameraQuirk, Boolean> quirksToChange = request.quirksToChange;
             String cameraUniqueName = request.cameraUniqueName;
 
             if (cameraUniqueName == null || cameraUniqueName.isEmpty()) {

--- a/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
+++ b/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
@@ -1171,7 +1171,7 @@ public class RequestHandler {
     }
 
     public static void onImageSnapshotsRequest(Context ctx) {
-        var snapshots = new ArrayList<HashMap<String, Object>>();
+        var snapshots = new ArrayList<Map<String, Object>>();
         var cameraDirs = ConfigManager.getInstance().getImageSavePath().toFile().listFiles();
 
         if (cameraDirs != null) {
@@ -1209,19 +1209,18 @@ public class RequestHandler {
 
     public static void onCameraCalibImagesRequest(Context ctx) {
         try {
-            HashMap<String, HashMap<String, ArrayList<HashMap<String, Object>>>> snapshots =
-                    new HashMap<>();
+            Map<String, Map<String, ArrayList<Map<String, Object>>>> snapshots = new HashMap<>();
 
             var cameraDirs = ConfigManager.getInstance().getCalibDir().toFile().listFiles();
             if (cameraDirs != null) {
-                var camData = new HashMap<String, ArrayList<HashMap<String, Object>>>();
+                var camData = new HashMap<String, ArrayList<Map<String, Object>>>();
                 for (var cameraDir : cameraDirs) {
                     var resolutionDirs = cameraDir.listFiles();
                     if (resolutionDirs == null) continue;
                     for (var resolutionDir : resolutionDirs) {
                         var calibImages = resolutionDir.listFiles();
                         if (calibImages == null) continue;
-                        var resolutionImages = new ArrayList<HashMap<String, Object>>();
+                        var resolutionImages = new ArrayList<Map<String, Object>>();
                         for (var calibImg : calibImages) {
                             var snapshotData = new HashMap<String, Object>();
 

